src/shogun/base/unique.h

```diff
@@ -0,0 +1,44 @@
+#ifndef __SG_UNIQUE_H__
+#define __SG_UNIQUE_H__
+
+#ifdef HAVE_CXX11
+
+#include <memory>
+
+namespace shogun
```

Comment:
Because of the `std::unique_ptr` use in the header. Also, this is to be used internally, not something to expose, right?